src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -226,57 +227,92 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
-  def adminDeleteWorkspace(workspaceName: WorkspaceName): Future[PerRequestMessage] = asFCAdmin {
+  def adminGetWorkspaceContext(workspaceName: WorkspaceName): Future[SlickWorkspaceContext] = {
     dataSource.inTransaction { dataAccess =>
       withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
-        deleteWorkspace(workspaceName, dataAccess, workspaceContext)
+        DBIO.successful(workspaceContext)
       }
     }
   }
 
-  def deleteWorkspace(workspaceName: WorkspaceName): Future[PerRequestMessage] =
+  def getWorkspaceContext(workspaceName: WorkspaceName): Future[SlickWorkspaceContext] = {
     dataSource.inTransaction { dataAccess =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Owner, dataAccess) { workspaceContext =>
-        deleteWorkspace(workspaceName, dataAccess, workspaceContext)
+        DBIO.successful(workspaceContext)
       }
     }
+  }
+
+  def adminDeleteWorkspace(workspaceName: WorkspaceName): Future[PerRequestMessage] = asFCAdmin {
+    adminGetWorkspaceContext(workspaceName) flatMap { ctx =>
```

Comment:
I'd just do `asFCAdmin { deleteWorkspace() }` (using the function immediately below) here.